cache/rediscache/rediscache.go

```diff
@@ -0,0 +1,114 @@
+package rediscache
+
+import (
+	"time"
+
+	"github.com/go-redis/redis"
+	"github.com/terranodo/tegola/cache"
+	"github.com/terranodo/tegola/util/dict"
+)
+
+const CacheType = "redis"
+
+const (
+	ConfigKeyNetwork  = "network"
+	ConfigKeyAddress  = "address"
+	ConfigKeyPassword = "password"
+	ConfigKeyDB       = "db"
+	ConfigKeyMaxZoom  = "max_zoom"
+)
+
+func init() {
+	cache.Register(CacheType, New)
+}
+
+func New(config map[string]interface{}) (rcache cache.Interface, err error) {
+
+	// default values
+	defaultNetwork := "tcp"
+	defaultAddress := "127.0.0.1:6379"
+	defaultPassword := ""
+	defaultDB := 0
+	defaultMaxZoom := 0
+
+	c := dict.M(config)
+
+	network, err := c.String(ConfigKeyNetwork, &defaultNetwork)
+	if err != nil {
+		return
```

Comment:
Since this a pretty long function, please don't use a bare return. 
Especially since it is easy to do a 
`return nil, nil` 
here.